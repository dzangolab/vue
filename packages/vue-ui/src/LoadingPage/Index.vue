<template>
  <div class="loading-page">
    <LoadingIcon class="loading-icon" />
  </div>
</template>

<script lang="ts">
export default {
  name: "LoadingPage",
};
</script>

<script setup lang="ts">
import LoadingIcon from "../components/LoadingIcon.vue";
</script>

<style lang="css">
.loading-page {
  align-items: center;
  background: rgba(255, 255, 255, 0.6);
  display: flex;
  inset: 0;
  justify-content: center;
  position: fixed;
  z-index: 999;
}

.loading-page .loading-icon {
  color: black;
  font-size: 0.5rem;
}
</style>
