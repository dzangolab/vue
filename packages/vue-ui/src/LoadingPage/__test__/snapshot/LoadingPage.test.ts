import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import LoadingPage from "../../Index.vue";

describe("LoadingPage", () => {
  it("matches snapshot", () => {
    const wrapper = mount(LoadingPage, {
      props: {
        loading: true,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
