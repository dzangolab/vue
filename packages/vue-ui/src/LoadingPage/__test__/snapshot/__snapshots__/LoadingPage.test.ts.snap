// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingPage > matches snapshot 1`] = `
<div
  class="loading-page"
>
  <div
    class="loading loading-icon"
    data-v-20744668=""
  >
    Loading
  </div>
</div>
`;
